[Teams] add device posture docs (#1844)
cloudflare/cloudflare-docs: products/cloudflare-one/src/content/identity/devices/file-check.md

---
order: 12
pcx-content-type: how-to
---

# File Check

<Aside type="note">

> You can create multiple instances of the File Check for each operative system you need to run it on, or if you need to check for multiple files.

Reviewer: ```suggestion
You can create multiple instances of the File Check for each operating system you need to run it on, or if you need to check for multiple files.
```

</Aside>

The File Check device posture attribute checks for the presence of a file on a device.

| Prerequisites | 
| ------------- |
| This check is supported on: Windows, macOS, Linux|
| This check is not supported on: Chrome, Android, or iOS |

To enable File Check:

1. On the Teams Dashboard, navigate to **My Team** > **Devices** > **Device posture**.
1. Click **+Add**.
1. Select **File Check**.
1. Enter a descriptive name for the check.
1. Select your operating system.

> 1. Enter a file path (ex: `c:\my folder\myfile.exe`).

Reviewer: ```suggestion
1. Enter a file path (for example, `c:\my folder\myfile.exe`).
```

1. Next, you can choose to enter a [**Signing certificate thumbprint**](#determine-the-signing-thumbprint). 

>     Adding this information will enable the check to ensure that the file was signed by the software developer you expect it to be signed from. 

Reviewer: Same recommendation as above:
```suggestion
    Adding this information will enable the check to ensure that the application was signed by the expected software developer.
```

> 1. You can also opt to enter a [**SHA-256** value](#determine-the-sha-256-value). 

Reviewer: Add a new line before the new step.

>     This enables the check to ensure that the exact binary is on the device.

Reviewer: ```suggestion
    This enables the check to ensure the integrity of the binary file on the device.
```

1. Click **Save**.

Your device posture attribute is now visible on the **Device posture** page.

## Determine the signing thumbprint

> The process to determine the signing thumbprint of a file varies depending on the operative system. This is how you would look up the signing thumbrpint of the Cloudflare WARP application on macOS and Windows.

Reviewer: ```suggestion
The process to determine the signing thumbprint of a file varies depending on the operating system. This is how you would look up the signing thumbprint of the Cloudflare WARP application on macOS and Windows.
```

### On macOS

1. Create a directory.

    ```bash

>     ~/Desktop % mkdir tmp

Reviewer: Same comment as before about `$` and `sh` syntax highlighting.

    ```

>     ```bash

Reviewer: Either add an empty line between code blocks, or include the two commands in the same code block (separated by an empty line).

    ~/Desktop % cd tmp
    ```

1. Run the following command to extract certificates for the WARP application:

    ```bash
    ~/Desktop/tmp % codesign -d --extract-certificates "/Applications/Cloudflare WARP.app/Contents/Resources/CloudflareWARP"
    Executable=/Applications/Cloudflare WARP.app/Contents/Resources/CloudflareWARP
    ```

> 1. Next, run the following commant to extract the `SHA1` thumbprint:

Reviewer: Check formatting of `` `SHA1` `` as before.

    ```bash
    ~/Desktop/tmp % openssl x509 -inform DER -in codesign0 -fingerprint -sha1 -noout | tr -d :                              
    SHA1 Fingerprint=FE2C359D79D4CEAE6BDF7EFB507326C6B4E2436E
    ```

### On Windows

1. Open a PowerShell window.
1. Use the `Get-AuthenticodeSignature` command to find the thumbprint. For example:

```txt
Get-AuthenticodeSignature -FilePath c:\myfile.exe
```

## Determine the SHA-256 value

The SHA-256 value almost always changes between versions of a file/application. 

### On macOS

1. Open a Terminal window.
1. Use the `shasum` command to find the sha256 value of the file. For example:

```bash
$ shasum -a 256 myfile
```

### On Windows

1. Open a PowerShell window.
1. Use the `get-filehash` command to find the sha256 value of the file. For example:

> ```

Reviewer: ```suggestion
```txt
```

> get-filehash -path "C:\myfile.exe" -Algorithm SHA256 | format-list`)

Reviewer: ```suggestion
get-filehash -path "C:\myfile.exe" -Algorithm SHA256 | format-list
```
I believe these are extra characters that should be removed.
